src/DDTrace/Integrations/ElasticSearch/V1/ElasticSearchIntegration.php

```diff
@@ -17,6 +17,22 @@ class ElasticSearchIntegration extends AbstractIntegration
     const NAME = 'elasticsearch';
     const DEFAULT_SERVICE_NAME = 'elasticsearch';
 
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
```

Comment:
Would it make sense to move this up to the abstract `Integration` class since nearly all the integrations need it? Also - what do you think about making the `Integration` class constructor private to help enforce singleton? :)

Reply:
Sounds good! Originally I did this because classes are not marked as final and making the constructor protect the constructor meant breaking things for users that possibly extended them. At this point I am both moving to parent class instantiation and making classes final. 👍